proxy/coco/src/peer/run_state.rs

```diff
@@ -199,29 +284,59 @@ impl From<Config> for RunState {
             config,
             connected_peers: HashSet::new(),
             status: Status::Stopped(Instant::now()),
+            waiting_room: WaitingRoom::new(waiting_room::Config::default()),
         }
     }
 }
 
 impl RunState {
     /// Constructs a new state.
     #[cfg(test)]
-    const fn new(config: Config, connected_peers: HashSet<PeerId>, status: Status) -> Self {
+    fn new(config: Config, connected_peers: HashSet<PeerId>, status: Status) -> Self {
         Self {
             config,
             connected_peers,
             status,
+            waiting_room: WaitingRoom::new(waiting_room::Config::default()),
```

Comment:
But then this one is safe I would say because it's useful to have no delta during testing.